### DISCUSSION TITLE
Connect-PnpOnline fails with Authentication Context and MFA

### DISCUSSION THREAD

**Discussion by BenHagueNZ** [pnp/powershell#2297] [Q&A]
I have a SharePoint site protected by an authentication context that requires MFA. Connecting to that site with Connect-PnpOnline appears to work, but trying to do Get-PnpList fails with a (403) Forbidden error. Excluding the user from the conditional access policy requiring MFA and reconnecting allows Get-PnpList to work as normal. I've tried -UseWebLogin, -Interactive, and -DeviceLogin, all fail in the same way. On a PC that's joined to Azure AD I can get round this by first connecting to the site using Word and then connecting with Connect-PnpOnline. Is this not supported until authentication contexts are out of preview?

**Comment by FabrizioRibeiro** [comment]
Have you tried -PnPManagementShell?

**Reply by BenHagueNZ** [reply]
This is with the Pnp Management Shell enterprise application installed in the tenant.
